cmd/azure.go

```diff
@@ -38,9 +39,35 @@ import (
 
 const globalAzureAPIVersion = "2016-05-31"
 
+// To store metadata during NewMultipartUpload which will be used after
+// CompleteMultipartUpload to call SetBlobMetadata.
+type azureMultipart struct {
+	meta map[string]map[string]string
+	mu   sync.Mutex
```

Comment:
Why not use NSlock in respective calls ? Avoids having more locks here which are not needed ?.